nemo/collections/asr/models/ctc_models.py

```diff
@@ -79,7 +80,28 @@ def setup_test_data(self, test_data_layer_params: Optional[Dict]):
         self.__test_dl = self.__setup_dataloader_from_config(config=test_data_layer_params)
 
     def setup_optimization(self, optim_params: Optional[Dict]):
-        self.__optimizer = torch.optim.Adam(self.parameters(), lr=optim_params['lr'])
+        optim_params = optim_params or {}  # In case null was passed as optim_params
+
+        # Check if caller provided optimizer name, default to Adam otherwise
+        optimizer_name = optim_params.get('optimizer', 'adam')
+
+        # Check if caller has optimizer kwargs, default to empty dictionary
+        optimizer_args = optim_params.get('opt_args', [])
+        optimizer_args = parse_optimizer_args(optimizer_args)
+
+        # We are guarenteed to have lr since it is required by the argparser
+        # But maybe user forgot to pass it to this function
+        lr = optim_params.get('lr', None)
+
+        if 'lr' is None:
+            raise ValueError('`lr` must be passed when setting up the optimization !')
+
+        # Actually instantiate the optimizer
+        optimizer = get_optimizer(optimizer_name)
+        self.__optimizer = optimizer(self.parameters(), lr=lr, **optimizer_args)
```

Comment:
is this where it will throw if wrong optimizer_args are passed? E.g. betas for SGD?